src/ConnectionAutomaton.cpp

```diff
@@ -60,8 +60,7 @@ void ConnectionAutomaton::connect(
     std::shared_ptr<FrameTransport> frameTransport,
     bool sendingPendingFrames) {
   CHECK(isDisconnectedOrClosed());
-  CHECK(frameTransport);
-  CHECK(!frameTransport->isClosed());
+  CHECK(frameTransport && !frameTransport->isClosed());
```

Comment:
There's an advantage to leaving this separate—we'll get better crash reports if the condition is ever violated.